dhis-2/dhis-api/src/main/java/org/hisp/dhis/i18n/I18nFormat.java

```diff
@@ -271,45 +255,42 @@ public String formatPeriod(Period period, boolean shortVersion) {
 
     PeriodType periodType = period.getPeriodType();
     String typeName = periodType.getName();
+    Calendar calendar = PeriodType.getCalendar();
+
+    if (periodType instanceof WeeklyAbstractPeriodType) {
+      DateTimeUnit start;
+      DateTimeUnit end;
+      String week = null;
+
+      if (calendar.isIso8601()) {
+        start = DateTimeUnit.fromJdkDate(period.getStartDate());
+        end = DateTimeUnit.fromJdkDate(period.getEndDate());
+        LocalDate startDate =
+            Instant.ofEpochMilli(period.getStartDate().getTime())
+                .atZone(ZoneId.systemDefault())
+                .toLocalDate();
+        WeekFields weekFields =
+            WeekFields.of(PeriodType.MAP_WEEK_TYPE.get(periodType.getName()), 4);
+        week = String.valueOf(startDate.get(weekFields.weekOfWeekBasedYear()));
 
-    if (periodType instanceof WeeklyAbstractPeriodType) // Use ISO dates
-    // due to
-    // potential week
-    // confusion
-    {
-      DateTime dateTime = new DateTime(period.getStartDate());
-      LocalDate startDate =
-          Instant.ofEpochMilli(period.getStartDate().getTime())
-              .atZone(ZoneId.systemDefault())
-              .toLocalDate();
-      LocalDate endDate =
-          Instant.ofEpochMilli(period.getEndDate().getTime())
-              .atZone(ZoneId.systemDefault())
-              .toLocalDate();
-      WeekFields weekFields = WeekFields.of(PeriodType.MAP_WEEK_TYPE.get(periodType.getName()), 4);
-      String year = String.valueOf(startDate.get(weekFields.weekBasedYear()));
-      String week = String.valueOf(startDate.get(weekFields.weekOfWeekBasedYear()));
-
-      if (isWeeklyPeriodType(periodType)) {
-        return String.format(
-            shortVersion
-                ? "W%s %d-%02d-%02d - %d-%02d-%02d"
-                : "Week %s %d-%02d-%02d - %d-%02d-%02d",
-            week,
-            startDate.getYear(),
-            startDate.getMonth().getValue(),
-            startDate.getDayOfMonth(),
-            endDate.getYear(),
-            endDate.getMonth().getValue(),
-            endDate.getDayOfMonth());
+      } else {
+        start = calendar.fromIso(period.getStartDate());
+        end = calendar.fromIso(period.getEndDate());
+        week = String.valueOf(calendar.week(start));
       }
 
-      year += dateTime.dayOfWeek().getAsShortText() + " " + year;
+      return String.format(
```

Comment:
Unclear: It looks as if we now always use this formatting, not just for weekly period types. Why is that? Is that a change in observable behavior?

Reply:
for other period types we are not using start and end dates in the name ... we just say 012025 for example Monthly period january 2025. For weeks saying W12025 ... leaves users into lots of guess for the exact week. Hence the need for start and end dates based on user calendar

Reply:
yes, that is how it is currently. the changed code for name formatting sits under 
`if (periodType instanceof WeeklyAbstractPeriodType) `